pyreisejl/utility/converters.py

```diff
@@ -1,132 +1,51 @@
-        ],
-        inplace=True,
-    )
-    mpc["mpc"]["branch"] = branch.values
-    mpc["mpc"]["branchid"] = branchid
-    mpc["mpc"]["branchdevicetype"] = branchdevicetype
-
-    # generation cost
-    gencost = grid.gencost.copy()
-    gencost["before"].reset_index(inplace=True, drop=True)
-    gencost["before"].drop(columns=["interconnect"], inplace=True)
-    mpc["mpc"]["gencost"] = gencost["before"].values
-
-    # DC line
-    if len(grid.dcline) > 0:
-        dcline = grid.dcline.copy()
-        dclineid = dcline.index.values[np.newaxis].T
-        dcline.reset_index(inplace=True, drop=True)
-        dcline.drop(columns=["from_interconnect", "to_interconnect"], inplace=True)
-        mpc["mpc"]["dcline"] = dcline.values
-        mpc["mpc"]["dclineid"] = dclineid
-
-    # energy storage
-    mpc_storage = None
-
-    if len(grid.storage["gen"]) > 0:
-        storage = grid.storage.copy()
-
-        mpc_storage = {
-            "storage": {
-                "xgd_table": np.array([]),
-                "gen": np.array(storage["gen"].values, dtype=np.float64),
-                "sd_table": {
-                    "colnames": storage["StorageData"].columns.values[np.newaxis],
-                    "data": storage["StorageData"].values,
-                },
-            }
-        }
-
-    if filepath is not None:
-        savemat(filepath, mpc, appendmat=False)
-        if mpc_storage is not None:
-            savemat(storage_filepath, mpc_storage, appendmat=False)
-
-    return mpc, mpc_storage
+import os
+import pickle
+
+cols = {
+    "branch": ["branch_id", "from_bus_id", "to_bus_id", "x", "rateA"],
+    "dcline": ["dcline_id", "from_bus_id", "to_bus_id", "Pmin", "Pmax"],
+    "bus": ["bus_id", "Pd", "zone_id"],
+    "plant": [
+        "plant_id",
+        "bus_id",
+        "status",
+        "Pmin",
+        "Pmax",
+        "type",
+        "ramp_30",
+        "GenFuelCost",
+        "GenIOB",
+        "GenIOC",
+        "GenIOD",
+    ],
+}
+
+drop_cols = {"gencost": ["plant_id", "interconnect"]}
+drop_cols["gencost_orig"] = drop_cols["gencost"]
+
+
+def _save(path, name, df):
+    df = df.reset_index()
+    df = df.loc[:, cols.get(name, df.columns)]
+    df = df.drop(drop_cols.get(name, []), axis=1)
+    df.to_csv(os.path.join(path, f"{name}.csv"), index=False)
+
+
+def _save_storage(path, name, df):
+    df.to_csv(os.path.join(path, f"{name}.csv"), index=False)
+
+
+def pkl_to_csv(path):
+    with open(os.path.join(path, "grid.pkl"), "rb") as f:
+        grid = pickle.load(f)
+    _save(path, "branch", grid.branch)
+    _save(path, "dcline", grid.dcline)
+    _save(path, "bus", grid.bus)
+    _save(path, "plant", grid.plant)
+    _save(path, "gencost_orig", grid.gencost["before"])
+    _save(path, "gencost", grid.gencost["after"])
+
+    storage = grid.storage
+    if not storage["gen"].empty:
+        _save_storage(path, "StorageData", storage["StorageData"])
+        _save_storage(path, "storage_gen", storage["gen"])
```

Comment:
@lanesmith I'm wondering how a scenario with demand flexibility enabled looks like at this stage. Currently, we have `demand_flexibility_parameters.csv` passed through with corresponding profiles in the input folder, right? Those are additional information that a scenario object carries other than `start date`, `end date` and `interval`. Should we add some checks in `call.py` regarding that as well?

Reply:
Correct. All of the information pertaining to a scenario with demand flexibility enabled is contained within .csv files (e.g., `demand_flexibility_parameters.csv`, `demand_flexibility_up.csv`, etc.). Since that information is passed as .csv files, there is nothing that needs to be done to those files in this stage (i.e., `pyreisejl/utility/converters.py`). There are some checks that occur in `read.jl` for the different demand-flexibility-related files. What kind of checks did you have in mind?

Reply:
Idk. Currently, all demand flexibility related checks are run in Julia side, not sure whether we should port it out to Python side or even `PowerSimData` (or have them in both ends), similarly as what we have done with `storage`. Or this feature so far is still engine dependent, i.e. `REISE.jl` only, and we should keep it in this way. Just want to get your thoughts, no concrete ideas yet.

Reply:
I don't think there are any demand-flexibility-related checks that are engine dependent. Since most of the demand-flexibility-related information is profile-based and isn't really adjusted by `PowerSimData` (as of now), there hasn't really been much of a need to read those files and check that data in `PowerSimData`. I believe the other demand flexibility parameters have minor checks in `PowerSimData`, but I might be misremembering. I know we've talked about making the integration between `REISE.jl` and `PowerSimData` tighter (i.e., having `REISE.jl` be something that is only run through `PowerSimData` and not have it be run in the stand-alone version). Until we implement that, I think it makes sense to keep the checks in `REISE.jl` so that errors can be caught no matter how the engine is run.

Reply:
By engine dependent I mean only `REISE.jl` can handle demand flexibility (eg, `REISE`, `PowerSimulationsWrapper.jl`, `PypsaWrapper`, etc), thus all related logic should be done in `REISE.jl` at this point and there is no need to factor out  engine-independent part, such as checks, into `PowerSimData`.